addOns/automation/src/test/java/org/zaproxy/addon/automation/jobs/ActiveScanJobUnitTest.java

```diff
@@ -447,14 +449,64 @@ void shouldReturnScanPolicyForDefaultData() throws MalformedURLException {
         job.verifyParameters(progress);
         ScanPolicy policy = job.getData().getPolicyDefinition().getScanPolicy(null, progress);
 
+        // Then
+        assertThat(policy, is(equalTo(null)));
+        assertThat(progress.hasWarnings(), is(equalTo(false)));
+        assertThat(progress.hasErrors(), is(equalTo(false)));
+    }
+
+    @ParameterizedTest
+    @EnumSource(
+            value = AttackStrength.class,
+            names = {"LOW", "MEDIUM", "HIGH", "INSANE"})
+    void shouldReturnScanPolicyIfOnlyDefaultStrength(AttackStrength attackStrength) {
+        // Given
+        ActiveScanJob job = new ActiveScanJob();
+        AutomationProgress progress = new AutomationProgress();
+        LinkedHashMap<String, LinkedHashMap<?, ?>> data = new LinkedHashMap<>();
+        LinkedHashMap<String, String> policyDefn = new LinkedHashMap<>();
+        policyDefn.put("defaultStrength", attackStrength.name());
+        data.put("policyDefinition", policyDefn);
+
+        // When
+        job.setJobData(data);
+        job.verifyParameters(progress);
+        ScanPolicy policy = job.getData().getPolicyDefinition().getScanPolicy(null, progress);
+
         // Then
         assertThat(policy, is(notNullValue()));
-        assertThat(policy.getDefaultStrength(), is(AttackStrength.MEDIUM));
+        assertThat(policy.getDefaultStrength(), is(attackStrength));
         assertThat(policy.getDefaultThreshold(), is(AlertThreshold.MEDIUM));
         assertThat(progress.hasWarnings(), is(equalTo(false)));
         assertThat(progress.hasErrors(), is(equalTo(false)));
     }
 
+    @ParameterizedTest
+    @EnumSource(
+            value = AlertThreshold.class,
+            names = {"OFF", "LOW", "MEDIUM", "HIGH"})
```

Comment:
More a nit, since there are less that we don't want than the ones we want it could use the `EXCLUDE` mode. https://junit.org/junit5/docs/current/user-guide/#writing-tests-parameterized-tests-sources-EnumSource